src/QsCompiler/CompilationManager/DiagnosticTools.cs

```diff
@@ -311,33 +249,29 @@ internal static bool SelectByEndLine(this Diagnostic m, int lowerBound, int uppe
         /// <summary>
         /// Returns true if the start position of the given diagnostic is larger or equal to lowerBound.
         /// </summary>
-        internal static bool SelectByStart(this Diagnostic m, Lsp.Position lowerBound)
-        {
-            return m?.Range?.Start?.Line == null ? false : lowerBound.IsSmallerThanOrEqualTo(m.Range.Start);
-        }
+        internal static bool SelectByStart(this Diagnostic m, Position lowerBound) =>
+            m?.Range?.Start?.Line != null && lowerBound <= m.Range.Start.ToQSharp();
 
         /// <summary>
         /// Returns true if the start position of the given diagnostic is larger or equal to lowerBound, and smaller than upperBound.
         /// </summary>
-        internal static bool SelectByStart(this Diagnostic m, Lsp.Position lowerBound, Lsp.Position upperBound)
-        {
-            return m?.Range?.Start?.Line == null ? false : lowerBound.IsSmallerThanOrEqualTo(m.Range.Start) && m.Range.Start.IsSmallerThan(upperBound);
-        }
+        internal static bool SelectByStart(this Diagnostic m, Position lowerBound, Position upperBound) =>
+            m?.Range?.Start?.Line != null
+            && lowerBound <= m.Range.Start.ToQSharp()
+            && m.Range.Start.ToQSharp() < upperBound;
```

Comment:
Isn't there already a method, I think on the new Range type, that checks if a Position is within the Range? It seems like that logic should be centralized.

Reply:
Hmm, I guess not. But maybe there should be?

Reply:
I added `Range.Contains` and rewrote these two functions.